src/MergeContext.js

```diff
@@ -454,6 +454,265 @@ class BranchPosition
     }
 }
 
+// Forward iterator for fields in the 'name:value' format.
+class FieldsTokenizer
+{
+    constructor(str, maxFieldNumber) {
+        this._lines = str.split('\n');
+        // if provided, limits the number of fields to parse
+        this._maxFieldNumber = maxFieldNumber;
+        this._remainingFields = [];
+        this._tokenizeAll();
+    }
+
+    // returns the next parsed field in the {name, value, raw} format
+    nextField() {
+        assert(!this.atEnd());
+        return this._remainingFields.shift();
+    }
+
+    // parses all input in advance
+    _tokenizeAll() {
+        while (this._lines.length) {
+            const line = this._lines.shift();
+
+            // treat an empty line as the end of input
+            if (line.trim().length === 0)
+                break;
+
+            if (this._maxFieldNumber && this._remainingFields.length == this._maxFieldNumber)
+                throw new Error(`exceeded the maximum number of fields: ${this._maxFieldNumber}`);
+
+            if (/^\s/.test(line))
+                throw new Error(`a field cannot start with whitespace: ${line}`);
+
+            const pos = line.search(': ');
+            if (pos < 0)
+                throw new Error(`a field without a name: value delimiter: '${line}'`);
+
+            const name = line.substring(0, pos);
+            const value = line.substring(pos+2).trim();
+            if (this._remainingFields.some(el => el.name.toUpperCase() === name.toUpperCase() &&
```

Comment:
We should probably ban field name with spaces, including trailing spaces.

Reply:
Done (965d814).